libbeat/statestore/backend/es/notifier.go

```diff
@@ -0,0 +1,65 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package es
+
+import (
+	"sync"
+
+	conf "github.com/elastic/elastic-agent-libs/config"
+)
+
+type OnConfigUpdateFunc func(c *conf.C)
+type UnsubscribeFunc func()
+
+type Notifier struct {
+	mx sync.RWMutex
```

Comment:
```suggestion
	mx sync.Mutex
```

Another perhaps overly cautious observation: The only purpose of a `RWMutex` is to allow calling `Notify()` concurrently. However, I don't see the benefit of that: `Notify()` should finish quite fast since it just starts goroutines and more importantly, concurrent calls of `Notify()` lead to a race condition between the same callback which _could_ mean that the wrong item gets stored due to timing issues.